### DISCUSSION TITLE
CLI Functions

### DISCUSSION THREAD

**Discussion by Inpacchi** [Inpacchi/Manga-Tagger#26] [Ideas]
# CLI Functions:
#### Ask the user if a delimiter has been set or no
1. Process a single manga chapter
   * Pass in filename
2. Process a single manga series
   * Pass in a directory
3. Process multiple manga series
   * Pass in a directory

# CLI Options:
1. Set delimiter
2. Set output directory
3. Set rename or not 
   * I.e if they just want the manga to be tagged isntead of renamed

**Comment by DevParapalli** [comment]
A Typer based CLI would be easier to implement if the main program uses functions with named params to do its job. A cli.py with helper functions and the actual Tyler code would work.

**Comment by Ablex222** [comment]
Not very familiar with backend of CLI but this is probably the routes that need to be taken.
1. Expand syntax (Ch., Chapter, Volume, V, Ranges (Ch. 1-5 and V. 1-3 (such as omnibuses))
2. Separate out the databases for preexisting metadata and new pulls.
3. Remove software's dependence on FMD to run in CLI (Most files wouldn't be new pulls)
4. Linux compatible package (FMD is the primary windows dependent software)
5. Custom database installation script (Fast install)
6. Remove dependence of 3rd party databases (Extreme, but a combination of .txt documents and a autogenerated series of folders each housing the custom .XML for each series could accomplish the same task)
7. In general the less dependencies the better (More likely to get hung up otherwise)
8. Should be capable of being run alongside other CLI software (Batch scriptable)
9. Ideally can be ran without installing anything (portable, single executable)
Overall it would need to run on a case by case basis. Probably the files that are to be processed should not be deleted upon completion of the task, in order to make sure there are no defects.
